src/components/common/GoodsItem/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import Image from '@components/common/Image';
+import Ranking from './Ranking';
+
+const IMAGE_SIZE_RANKING = 160;
+const IMAGE_SIZE_GOODS_ITEM = 250;
+const IMAGE_RADIUS = 4;
+
+export interface GoodsItemProps {
+  imageSrc: string;
+  subtitle: string;
+  title: string;
+  amount: number;
+  rankingIndex?: number;
+}
+
+export default function GoodsItem({ imageSrc, subtitle, title, amount, rankingIndex, ...rest }: GoodsItemProps) {
+  const imageSize = rankingIndex ? IMAGE_SIZE_RANKING : IMAGE_SIZE_GOODS_ITEM;
+
+  const renderRanking = () => rankingIndex && <Ranking {...{ rankingIndex }} />;
+
+  return (
+    <StyledGoodsItem rankingIndex={rankingIndex} {...rest}>
+      {renderRanking()}
```

Comment:
```suggestion
      {rankingIndex && <Ranking {...{ rankingIndex }} />}
```

위처럼 inline 할 수 있을텐데 컴포넌트 내부의 함수로 추출해야하는 이유는 어떤걸까요?

Reply:
가독성과 추후 유지보수하기 용이할 부분이라고 생각되서 해당 조건부 렌더링을 함수로 따로 분리하게되었습니다!

Reply:
관점이 다를 수 있는데 제가 가진 생각을 전달드려 볼게요. 컴포넌트 안에서 jsx를 반환하는 함수를 만들기 보다는 jsx로 인라인 하는게 위, 아래로 왔다 갔다 하지 않고 한 눈에 볼 수 있어서 쉽게 읽힌다점에서 가독성 측면에서는 더 나은 선택이라고 봐요. jsx를 반환하는 함수는 곧 컴포넌트로 추출될 수도 있다는 말인데 컴포넌트 안에서 매번 리렌더링시 재정의 되는 것도 좋은 방향일지 생각해볼만한 점이기도 합니다.

그리고 유지보수를 언급해주신 부분은 어떤 상황이 왔을 때 문제가 될 수 있을까요? 가독성, 유지보수성처럼 추상적이라 해석의 여지가 있는 단어를 사용할 때는 구체적인 맥락을 담아주셔야 서로의 다른 관점을 정렬하고 건설적인 논의를 할 수 있기에 한번쯤 생각해보시면 좋겠습니다.